### PR TITLE
 Add cluster.group and cluster.rule commands

### PR DESCRIPTION
**Pull request by dougm**

### CODE REVIEW

**Comment by dougm** [comment]
@embano1 with these commands, govc can create/update/remove cluster DRS rules.  tl;dr is in the USAGE.md updates, I can also provide pointers on how to convert UI steps to govc commands if needed.

File: govc/cluster/rule/create.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+func (cmd *create) Description() string {
+	return `Create cluster rule.
+
+Rules are not enabled by default, use the 'enable' flag to enable upon creation or cluster.rule.change after creation.
+
+One of '-affinity', '-anti-affinity' or '-vm-host' must be provided to specify the rule type.
+
+With '-affinity' or '-anti-affinity', at least 2 vm NAME arguments must be specified.
+
+With '-vm-host', use the '-vm-group' flag combined with the '-host-affine-group' and/or '-host-anti-affine-group' flags.
```
Comment (anfernee) [inline govc/cluster/rule/create.go]: `affine` and `anti-affine`, are they typo?

File: govc/flags/cluster.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	v.DatacenterFlag, ctx = NewDatacenterFlag(ctx)
+	ctx = context.WithValue(ctx, clusterFlagKey, v)
+	return v, ctx
+}
+
+func (f *ClusterFlag) Register(ctx context.Context, fs *flag.FlagSet) {
+	f.RegisterOnce(func() {
+		f.DatacenterFlag.Register(ctx, fs)
+
+		env := "GOVC_CLUSTER"
```
Comment (anfernee) [inline govc/flags/cluster.go]: define it as a const here: `govc/flags/client.go`?

File: govc/cluster/rule/create.go (hunk 1 above)
Comment (dougm) [reply govc/cluster/rule/create.go]: I just used the same names as the fields: http://pubs.vmware.com/vsphere-6-5/index.jsp#com.vmware.wssdk.apiref.doc/vim.cluster.VmHostRuleInfo.html

File: govc/flags/cluster.go (hunk 2 above)
Comment (dougm) [reply govc/flags/cluster.go]: The client const's are referenced in more than 1 func.  GOVC_CLUSTER is only referenced within this func, you'll see the same pattern in other flags:
```console
% git --no-pager grep -n -e env\ \:\=\ \"GOVC_ --
govc/flags/datacenter.go:61:		env := "GOVC_DATACENTER"
govc/flags/datastore.go:63:		env := "GOVC_DATASTORE"
govc/flags/debug.go:51:		env := "GOVC_DEBUG"
govc/flags/folder.go:54:		env := "GOVC_FOLDER"
govc/flags/host_system.go:61:		env := "GOVC_HOST"
govc/flags/network.go:58:		env := "GOVC_NETWORK"
govc/flags/resource_pool.go:54:		env := "GOVC_RESOURCE_POOL"
govc/flags/storage_pod.go:39:		env := "GOVC_DATASTORE_CLUSTER"
govc/flags/virtual_app.go:57:		env := "GOVC_VAPP"
govc/flags/virtual_machine.go:60:		env := "GOVC_VM"
govc/vm/guest/auth.go:54:	env := "GOVC_GUEST_LOGIN"
```

**Comment by embano1** [comment]
AWESOME! THX A TON!!!!

**Pull request merged**